sync/course-db.js

```diff
@@ -1215,6 +1227,30 @@ async function validateAssessment(assessment, questions) {
     );
   }
 
+  const validRoleNames = new Set();
+  (assessment.groupRoles || []).forEach((role) => {
+    validRoleNames.add(role.name);
+  });
+
+  (assessment.zones || []).forEach((zone) => {
+    (zone.questions || []).forEach((zoneQuestion) => {
+      (zoneQuestion.canView || []).forEach((roleName) => {
+        if (!validRoleNames.has(roleName)) {
+          errors.push(
+            `A zone question's "canView" permission contains the non-existent group role name "${roleName}".`
+          );
```

Comment:
@nwalters512 Should we specify in the sync errors which question ID failed the validation? If so, should we have separate code to handle questions with alternatives vs. without alternatives?